### DISCUSSION TITLE
How to pass Object prop to custom vue extension?

### DISCUSSION THREAD

**Discussion by Arturexe** [ueberdosis/tiptap#2858] [General Questions]
Hello, I'm trying to pass an object `state.object` to a custom `vue component` extension but when the `html` string is sent it only shows: 

```
<vue-component prop="[object "></vue-component>
```

which seems to be a broken `string` of the object. My `node-extension.js` looks like this:

```
import {Node, mergeAttributes} from '@tiptap/core'
import {VueNodeViewRenderer} from '@tiptap/vue-3'
import Component from './Component.vue'

export const VueComponent = Node.create({
    name: 'vueComponent',
    group: 'block',
    draggable: true,
    addAttributes() {
        return {
            prop: Object
        }
    },
    parseHTML() {
        return [{ tag: 'vue-component' }]
    },
    renderHTML({ HTMLAttributes }) {
        return ['vue-component', mergeAttributes(HTMLAttributes)]
    },
    addNodeView() {
        return VueNodeViewRenderer(Component)
    },
})
```

**Comment by karmaral** [comment]
Do you want to have a separate attribute for each prop inside `Object` or just the plain contents?
If you want a separate attribute (`attr1=val1 attr2=val2`) you should probably return it like
```
addAttributes() {
  return { ...Object }
}
```
If there's just a single thing inside `Object`, maybe `prop: JSON.stringify(Object)` ?

I might be way off, but perhaps it helps.

**Reply by yaskevich** [reply]
Is it right that there is no way to actually _pass the props_ while one can _set the attributes_?

The crucial difference is that all attributes get in into JSON when serialized which may be undesirable, for example, for storing Tiptap content in a database.

**Comment by sereneinserenade** [comment]
have u tried using Pinia/Vuex/composite functions for the data sharing instead of passing props?
